Sources/MobilliumIntroduction/Classes/IntroductionController.swift

```diff
@@ -0,0 +1,236 @@
+//
+//  IntroductionController.swift
+//  MobilliumIntroduction
+//
+//  Created by Ahmet İmirze on 12.04.2022.
+//
+
+import UIKit
+
+// MARK: - IntroductionControllerPageDelegate
+public protocol IntroductionControllerPageDelegate: AnyObject {
+    func introductionController(_ controller: IntroductionController, didMoveToPage index: Int)
+}
+
+// MARK: - IntroductionControllerButtonDelegate
+public protocol IntroductionControllerButtonDelegate: AnyObject {
+    func didSkipButtonTapped()
+    func didNextButtonTappedAtEndOfContents()
+}
+
+// MARK: - IntroductionController
+public class IntroductionController: UIViewController {
+    
+    public weak var pageDelegate: IntroductionControllerPageDelegate?
+    public weak var buttonDelegate: IntroductionControllerButtonDelegate?
+    
+    private let pageControl: UIPageControl = {
+        let pageControl = UIPageControl()
+        pageControl.translatesAutoresizingMaskIntoConstraints = false
+        return pageControl
+    }()
+    
+    private let skipButton: UIButton = {
+        let button = UIButton()
+        button.translatesAutoresizingMaskIntoConstraints = false
+        return button
+    }()
+    
+    private let collectionView: UICollectionView = {
+        let layout = UICollectionViewFlowLayout()
+        layout.scrollDirection = .horizontal
+        let collectionView = UICollectionView(frame: .zero, collectionViewLayout: layout)
+        collectionView.backgroundColor = .clear
+        collectionView.register(IntroductionCell.self, forCellWithReuseIdentifier: "IntroductionCell")
+        collectionView.isPagingEnabled = true
+        collectionView.showsVerticalScrollIndicator = false
+        collectionView.showsHorizontalScrollIndicator = false
+        collectionView.translatesAutoresizingMaskIntoConstraints = false
+        return collectionView
+    }()
+    
+    private let nextButton: UIButton = {
+        let button = UIButton()
+        button.translatesAutoresizingMaskIntoConstraints = false
+        return button
+    }()
+    
+    private let config: IntroductionConfig
+    
+    public init(config: IntroductionConfig = IntroductionConfig()) {
+        self.config = config
+        super.init(nibName: nil, bundle: nil)
+        self.modalTransitionStyle = .coverVertical
+        self.modalPresentationStyle = .fullScreen
+    }
+    
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    public override func viewDidLoad() {
+        super.viewDidLoad()
+        view.backgroundColor = .white
+        addSubviews()
+        configureContents()
+    }
+    
+    @objc
+    private func skipButtonTapped(_ button: UIButton) {
+        buttonDelegate?.didSkipButtonTapped()
+        dismiss(animated: true)
```

Comment:
```suggestion
        dismiss(animated: true) {
        buttonDelegate?.didSkipButtonTapped()
        }
```
delegate dismiss closure içinde verilmesi daha iyi olabilir